### ISSUE TITLE
Port learning rate schedulers from Huggingface

### ISSUE THREAD

**Issue by maksym-del** [allenai/allennlp#4044]
Huggingface contains a bunch of custom ```LambdaLR``` based LR schedulers they use to get scores for different tasks.
We could simply port them so they can be instantiated from the configuration file. 

Here is a working example of how to register one of the schedulers:
```python

from typing import Any, Dict, List, Union

from allennlp.common.registrable import Registrable
from allennlp.training.optimizers import Optimizer
from allennlp.training.learning_rate_schedulers import LearningRateScheduler
from allennlp.training.learning_rate_schedulers.learning_rate_scheduler import  _PyTorchLearningRateSchedulerWrapper

from transformers import get_linear_schedule_with_warmup

@LearningRateScheduler.register("linear_with_warmup")
class WarmpuLinearLearningRateScheduler(_PyTorchLearningRateSchedulerWrapper):
    """
    Registered as a `LearningRateScheduler` with name "linear_with_warmup".
    """

    def __init__(
        self, optimizer: Optimizer, num_warmup_steps: int, num_training_steps: int, last_epoch: int = -1
    ) -> None:
        lr_scheduler = get_linear_schedule_with_warmup(
            optimizer=optimizer, 
            num_warmup_steps=num_warmup_steps, 
            num_training_steps=num_training_steps, 
            last_epoch=last_epoch)
        super().__init__(lr_scheduler)
```
If you like the idea, do you think we should add these at the end of the ```allennlp.training.learning_rate_schedulers``` file, or create a separate file (e.g. ``` huggingface_learning_rate_schedulers```)?

The function  ```get_linear_schedule_with_warmup``` returns regular pytroch ```lr scheduler```.

**Comment by matt-gardner** [comment]
Yes, I think this is a good idea. I'm ok with either putting them in [here](https://github.com/allenai/allennlp/blob/master/allennlp/training/learning_rate_schedulers/learning_rate_scheduler.py) or putting them in a separate file, with a slight preference towards the first option.
